### PR TITLE
VIM-3704: Demo project simplification

### PR DESCRIPTION
**Pull request by alfiehanssen**
#### Ticket
**Required for Vimeans only**
[VIM-3704](https://vimean.atlassian.net/browse/VIM-3704)

#### Ticket Summary
The README should more clearly describe how to use old and new upload. Both example projects should work as expected. And the VimeoUpload object should be genericized to accommodate both old and new upload.

#### Implementation Summary
Made VimeoUpload class generic. Cleaned up and squashed some bugs in the demo projects. Updated the README.

#### How to Test
Drop and auth token into each example project and ensure that uploads are working. Peep the README and confirm that it makes sense.

### CODE REVIEW

**Comment by alfiehanssen** [comment]
@huebnerob You have been chosen! :skull:

File: Example/VimeoUpload-iOS-2Step/VimeoUpload-iOS-2Step/AppDelegate.swift [hunk 1]
```diff
@@ -34,7 +34,8 @@ class AppDelegate: UIResponder, UIApplicationDelegate
     func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool
     {
         AFNetworkReachabilityManager.sharedManager().startMonitoring()
-        
+        NewVimeoUpload.sharedInstance // Ensure init is called on launch
```
Comment (huebnerob) [inline Example/VimeoUpload-iOS-2Step/VimeoUpload-iOS-2Step/AppDelegate.swift]: I don't like the idea of sharedInstance initializers with side effects, perhaps add a method like `setup()`?

File: Example/VimeoUpload-iOS-Example/VimeoUpload-iOS-Example/AppDelegate.swift [hunk 2]
```diff
@@ -34,15 +34,19 @@ class AppDelegate: UIResponder, UIApplicationDelegate
     func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool
     {
         AFNetworkReachabilityManager.sharedManager().startMonitoring()
-        
+        OldVimeoUpload.sharedInstance // Ensure init is called on launch
```
Comment (huebnerob) [inline Example/VimeoUpload-iOS-Example/VimeoUpload-iOS-Example/AppDelegate.swift]: :skull:

File: README.md [hunk 3]
```diff
@@ -295,85 +291,48 @@ This is quite a bit simpler:
...
 
 ```Swift
     let vimeoUpload = ... // Your instance of VimeoUpload (see above)
     let url = ... // Your url (see above)
-    let uploadTicket = ... // Your upload ticket (see above)
     
-    vimeoUpload.uploadVideo(url: url, uploadTicket: uploadTicket)
+    let title = "Untitled"
+    let description = "A really cool video"
+    let privacy = "nobody"
```
Comment (huebnerob) [inline README.md]: don't we have constants for privacy?

File: VimeoUpload/VimeoUpload.swift [hunk 4]
```diff
@@ -26,13 +26,13 @@
...
 import Foundation
 
-class VimeoUpload
+class VimeoUpload<T: VideoDescriptor>
 {
-    private static let Name = "vimeo_upload"
-    
     // MARK:
     
+    let name = "vimeo_upload" // Generic types don't yet support static properties [AH] 3/19/2016
```
Comment (huebnerob) [inline VimeoUpload/VimeoUpload.swift]: nit/fyi: could use a computed property instead... `static var name: String { return "vimeo_upload" }`

**Comment by huebnerob** [comment]
good stuff! :tada:

File: README.md (hunk 3 above)
Comment (alfiehanssen) [inline README.md]: Apparently not in the library itself, perhaps in the app?

File: Example/VimeoUpload-iOS-2Step/VimeoUpload-iOS-2Step/AppDelegate.swift (hunk 1 above)
Comment (alfiehanssen) [inline Example/VimeoUpload-iOS-2Step/VimeoUpload-iOS-2Step/AppDelegate.swift]: Good call, will do

**Pull request merged**